scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/Readme.md

```diff
@@ -0,0 +1,172 @@
+
+* --density: IPC density level (L,N,M) (default=`N`)
+* --ipc_doc: IPC definition document (default=`../ipc_definitions.yaml`)
+
+* --force_rectangle_pads: Force the generation of rectangle pads instead of rounded rectangle
+* --kicad4_compatible: Create footprints compatible with version 4 (avoids round-rect and custom pads).
+
+## Size Definition Format
+
+
+The size definition file contains one entry per package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was generated by which parameter set.)
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and Naming Parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- Footprint library name (`library`) {string}
+- Footprint name generation control
+  - Device type added as prefix. Example QFN. (`device_type`) {string}
+  - Selction of different name format. Example for LGA footprints. (`use_name_format`) {enum(`LGA`, `not LGA`)}
+  - Manufacturer name. Will be added as a prefix if given. (`manufacturer`) {string}
+  - Part number. Will be added as a prefix if given. (`part_number`) {string}
+  - Suffix: A custom suffix (added after pin pitch in default naming format). Can include parameters `pad_x` and `pad_y`.
+  - Custom naming (`custom_name_format`) {python format string}
+    - The full default format string is `{man:s}_{mpn:s}_{pkg:s}-{pincount:d}-1EP_{size_x:g}x{size_y:g}mm_P{pitch:g}mm{suffix:s}_EP{ep_size_x:g}x{ep_size_y:g}mm_Mask{mask_size_x:g}x{mask_size_y:g}mm{suffix2:s}{vias:s}` (The same parameters can be used in your custom format. Exposed pad parameters are not available for components without exposed pad.)
+
+_Note: Contributions intended for the official library shall not include the manufacturer or part number unless the footprint is specific to that manufacturer or part. Similarly avoid custom naming for official library contributions unless required to achieve the requested name (Example TI-specific naming)._
+
+### Package Dimensions
+![dimension example](./documentation/dimension_system.svg)
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Lead width (`lead_width`) {dimension}
+  - Lead length (`lead_len`) {dimension}
+    - Optional for specifying different length for horizontal and vertical leads (`lead_len_H`, `lead_len_V`) {dimension}
+- Lead pitch, currently equal for all sides (`pitch`) {float}
+
+#### Pull Back Leads
+By default the leads are assumed to extend to the body edge. Some components however have the leads pulled back a bit. A typical example are packages that are called LGA by manufacturers.
```

Comment:
To me, 'Some components however have the leads pulled back a bit.' reads a bit awkwardly. Perhaps 'On some packages the leads are inset within the package body and these are called "pull back" leads.' would be better?

Reply:
Fixed